### PR TITLE
UI Plugin: format UI code by prettier

### PR DESCRIPTION
**Pull request by raymondz1**
## What this PR does / why we need it

1. set up eslint and prettier
2. format all UI files.

## Which issue(s) this PR fixes

Fixes: #

### CODE REVIEW

File: cli/cmd/plugin/ui/web/tanzu-ui/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            "warn",
+            {
+            "code": 140,
+            "comments": 120,
+            "ignoreUrls": true
+            }
+        ],
+        "@typescript-eslint/no-explicit-any": "off",
+        "no-duplicate-imports": "warn",
+        "no-multiple-empty-lines": "error",
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/.eslintrc.json:47]: I'd prefer the multiple-empty-lines be a warning

File: cli/cmd/plugin/ui/web/tanzu-ui/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        ],
+        "operator-linebreak": [
+            "error",
+            "after"
+        ],
+        "quotes": [
+            "warn",
+            "single"
+        ],
+        "semi": [
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/.eslintrc.json:61]: I'd like to remove the "semi" as an error

File: cli/cmd/plugin/ui/web/tanzu-ui/.prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+    "semi": true,
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/.prettierrc.json:2]: If this adds semi-colons, I'd like to remove it

File: cli/cmd/plugin/ui/web/tanzu-ui/src/components/ClusterSettings.tsx [hunk 4]
```diff
@@ -41,15 +47,22 @@ function ClusterSettings(props: ChildProps | any) {
...
+                        defaultValue={state.data['CLUSTER_NAME'] || ''}
+                        onBlur={() => blurHandler()}
+                    />
+                    {errors['CLUSTER_NAME'] && <CdsControlMessage status="error">{errors['CLUSTER_NAME'].message}</CdsControlMessage>}
                 </CdsInput>
             </div>
-            <CdsButton onClick={() => {
-                handleFormSubmit({ ...props, submitForm });
-            }}>Next</CdsButton>
+            <CdsButton
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/components/ClusterSettings.tsx:59]: Whatever causes this change, I don't care for it. :>

File: cli/cmd/plugin/ui/web/tanzu-ui/src/components/aws/management-credential-step/management.credential.form.schema.ts [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
-    PROFILE: yup.string(),
-    REGION: yup.string().required(),
-    SECRET_ACCESS_KEY: yup.string(),
-    SESSION_TOKEN: yup.string(),
-    ACCESS_KEY_ID: yup.string(),
-    EC2_KEY_PAIR: yup.string().required()
-}).required();
\ No newline at end of file
+export const managementCredentialFormSchema = yup
+    .object({
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/components/aws/management-credential-step/management.credential.form.schema.ts:5]: Seems odd to put .object() on a separate line

File: cli/cmd/plugin/ui/web/tanzu-ui/src/state-management/reducers/Form.reducer.tsx [hunk 6]
```diff
@@ -15,14 +15,14 @@ interface FormState {
...
     switch (action.type) {
-    case INPUT_CHANGE:
-        newState = {
-            ...state,
-            [action.field]: action.payload,
-        };
-        break;
-    default:
-        newState = { ...state };
+        case INPUT_CHANGE:
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/state-management/reducers/Form.reducer.tsx:18]: I thought the (relatively new) standard was to have the `case` statements aligned with the `switch` statement

File: cli/cmd/plugin/ui/web/tanzu-ui/.eslintrc.json (hunk 2 above)
Comment (miclettej) [reply cli/cmd/plugin/ui/web/tanzu-ui/.eslintrc.json:61]: Agreed it doesn't need to be an error, but I would like to see it as a warning

File: cli/cmd/plugin/ui/web/tanzu-ui/package.json [hunk 7]
```diff
@@ -29,6 +29,7 @@
   },
   "scripts": {
     "lint": "eslint --ext js,ts,tsx src",
+    "format": "prettier --write src",
     "generate-api": "openapi --input ../../api/swagger.yaml --output ./src/swagger-api",
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/package.json:33]: Raymond, can you add a script named `pre-commit` that runs prettier and lint?

File: cli/cmd/plugin/ui/web/tanzu-ui/.eslintrc.json [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        "max-len": [
+            "warn",
+            {
+            "code": 140,
+            "comments": 140,
+            "ignoreUrls": true
+            }
+        ],
+        "@typescript-eslint/no-explicit-any": "off",
+        "no-duplicate-imports": "warn",
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/.eslintrc.json:46]: error

**Review by swalner-vmware — APPROVED** [review]

**Review by miclettej — APPROVED** [review]

**Pull request merged**